tests/auxiliary/user-account/test_contract.py

```diff
@@ -282,91 +286,107 @@ def test_call_checks_signature(vm):
         (27 if v == 28 else 28, r, SECPK1_N - s)
     ]
 
-    for invalid_vrs in invalid_vrs_triples:
-        transaction.vrs = invalid_vrs
-        assert transaction.is_signed
-
+    for v, r, s in invalid_vrs_triples:
+        transaction = UserAccountTransaction(
+            chain_id=transaction.chain_id,
+            shard_id=transaction.shard_id,
+            to=transaction.to,
+            gas=transaction.gas,
+            access_list=transaction.access_list,
+            destination=transaction.destination,
+            value=transaction.value,
+            nonce=transaction.nonce,
+            min_block=transaction.min_block,
+            max_block=transaction.max_block,
+            gas_price=transaction.gas_price,
+            msg_data=transaction.msg_data,
+            v=v,
+            r=r,
+            s=s,
+        )
+        # don't detect invalidity before transaction execution
+        transaction.validate = lambda: None
```

Comment:
I think a more appropriate way to check this would be to assemble a `Message` and then to run `vm.get_computation(message).apply_message()`.

This should execute the transaction *below* the place where the call to `transaction.validate()` occurs, allowing you to directly test that the EVM code appropriately validates the signature without having to do any monkeypatching of the transaction object.